lib/stb_image.ex

```diff
@@ -60,27 +67,31 @@ defmodule StbImage do
   ## Example
 
       {:ok, buffer} = File.read("/path/to/image")
-      {:ok, img, shape, type, channels} = StbImage.from_binary(buffer)
-      {h, w, c} = shape
+      stb_image_struct = StbImage.from_binary(buffer)
+      {h, w, c} = stb_image_struct.shape
+      img = stb_image_struct.data
 
       # If you know the image is a 4-channel image and auto-detection failed
-      {:ok, img, shape, type, channels} = StbImage.from_file("/path/to/image", channels: 4)
-      {h, w, c} = shape
+      stb_image_struct = StbImage.from_file("/path/to/image", channels: 4)
+      {h, w, c} = stb_image_struct.shape
+      img = stb_image_struct.data
 
   """
   def from_binary(buffer, opts \\ []) when is_binary(buffer) and is_list(opts) do
     type = opts[:type] || :u8
     channels = opts[:channels] || 0
-    StbImage.Nif.from_binary(buffer, channels, type)
+    {:ok, img, shape, type, channels} = StbImage.Nif.from_binary(buffer, channels, type)
+    {:ok, %StbImageWrapper{data: img, shape: shape, type: type, color_mode: channels}}
```

Comment:
You need to write this using `with` so it still returns `{:error, ...}` in case of invalid binaries:

```suggestion
    with {:ok, img, shape, type, channels} <- StbImage.Nif.from_binary(buffer, channels, type) do
      {:ok, %StbImageWrapper{data: img, shape: shape, type: type, color_mode: channels}}
    end
```

Please apply similar change to the other functions!